api/framework/Infrastructure/Identity/Users/Endpoints/ForgotPasswordEndpoint.cs

```diff
@@ -14,20 +14,15 @@ public static class ForgotPasswordEndpoint
 {
     internal static RouteHandlerBuilder MapForgotPasswordEndpoint(this IEndpointRouteBuilder endpoints)
     {
-        return endpoints.MapPost("/forgot-password", async (HttpRequest request, [FromHeader(Name = TenantConstants.Identifier)] string tenant, ForgotPasswordCommand command, IValidator<ForgotPasswordCommand> validator, IUserService userService, CancellationToken cancellationToken) =>
+        return endpoints.MapPost("/forgot-password", async (HttpRequest request, HttpContext context, [FromHeader(Name = TenantConstants.Identifier)] string tenant, ForgotPasswordCommand command, IValidator<ForgotPasswordCommand> validator, IUserService userService, CancellationToken cancellationToken) =>
         {
             ValidationResult result = await validator.ValidateAsync(command, cancellationToken);
             if (!result.IsValid)
             {
                 return Results.ValidationProblem(result.ToDictionary());
             }
 
-            // Obtain origin from request headers
-            var origin = request.Headers["Origin"].ToString();
-            if (string.IsNullOrEmpty(origin))
-            {
-                origin = "http://defaultOrigin.com"; // Default value if Origin header is not present
-            }
+            var origin = $"{context.Request.Scheme}://{context.Request.Host.Value}{context.Request.PathBase.Value}";
```

Comment:
This is still vulnerable to host header injection. My idea was to read the origin URL somewhere from a fixed location, rather than from the request. It would be better if we can define the origin URL in appsettings. This way no one can perform Host Header Injection / XSS. Thoughts?